x-pack/plugins/apm/common/rules/schema.ts

```diff
@@ -9,6 +9,16 @@ import { schema, TypeOf } from '@kbn/config-schema';
 import { ML_ANOMALY_SEVERITY } from '@kbn/ml-anomaly-utils/anomaly_severity';
 import { AggregationType, ApmRuleType } from './apm_rule_types';
 
+export const searchConfigurationSchema = schema.object({
+  query: schema.object({
+    query: schema.oneOf([
+      schema.string(),
+      schema.recordOf(schema.string(), schema.any()),
+    ]),
+    language: schema.string(),
```

Comment:
Would be better if we can specify the exact languages like:

```suggestion
    language: 'kuery' | 'eql' | 'lucene',
```

(again, this feels like this should be shared across Observability)

Reply:
I followed the same structure as [this](https://github.com/elastic/kibana/blob/1f00484d36c3f68cce400fdc649095f13f2255ec/packages/kbn-es-query/src/filters/build_filters/types.ts#L97) for consistency.